plugin/src/main/java/org/screamingsandals/bedwars/listener/PlayerListener.java

```diff
@@ -573,11 +573,11 @@ public void onFly(PlayerToggleFlightEvent event) {
         if (event.isCancelled()) {
             return;
         }
+
         Player player = event.getPlayer();
-        if (Main.isPlayerInGame(player)) {
-            if (!Main.getPlayerGameProfile(player).isSpectator) {
-                event.setCancelled(true);
-            }
+        if (Main.isPlayerInGame(player) && !Main.getPlayerGameProfile(player).isSpectator
+               && (!player.hasPermission("bw.bypass.flight") || Main.getConfigurator().config.getBoolean("disable-flight"))) {
```

Comment:
I would rather have this in separated check, it does not make that much sense like this :p

&& !player.hasPermission... && configurator..

That will make it like when player have the permission, it will not execute event.setCancelled()